cli/compilers/ts.rs

```diff
@@ -216,6 +217,22 @@ pub struct TsCompiler {
   pub compile_js: bool,
 }
 
+#[derive(Clone)]
+pub struct TsCompiler(Arc<TsCompilerInner>);
+
+impl Deref for TsCompiler {
+  type Target = TsCompilerInner;
+  fn deref(&self) -> &Self::Target {
+    &self.0
+  }
+}
+
+impl DerefMut for TsCompiler {
+  fn deref_mut(&mut self) -> &mut Self::Target {
+    Arc::get_mut(&mut self.0).unwrap()
```

Comment:
Are you sure you want to use Arc::get_mut() here? Note that the latter will fail if the Arc has > 1 references to it.
I suspect you really just want to use `Option<TSCompiler>` to store the thing and `as_mut()` to deref it.

Reply:
removed